Plugins/GeoModel/src/GeoModelToDetectorVolume.cpp

```diff
@@ -126,12 +126,25 @@ Volume convertVolume(const Transform3& trf, const GeoShape& shape) {
         dynamic_cast<const GeoShapeShift*>(&shape);
     const GeoShape* shapeOp = shiftShape->getOp();
     newTrf = trf * shiftShape->getX();
-    return convertVolume(newTrf, *shapeOp);
+    // This assumes, without checking, that newTrf is an isometry
+    Transform3 newtrf = Eigen::Isometry3d(newTrf.matrix());
```

Comment:
Also here, this won't work?

```suggestion
    Transform3 newtrf{newTrf.matrix()};
```

Reply:
Your suggestion works. Updated the code accordingly.